aloscene/scene_flow.py

```diff
@@ -0,0 +1,182 @@
+    Scene flow map
+
+    Parameters
+    ----------
+    x : str
+        load scene flow from a numpy file
+    """
+
+    @staticmethod
+    def __new__(cls, x, occlusion: Mask = None, *args, names=("C", "H", "W"), **kwargs):
+        if isinstance(x, str):
+            # load flow from path
+            x = load_scene_flow(x)
+            names = ("C", "H", "W")
+
+        tensor = super().__new__(cls, x, *args, names=names, **kwargs)
+        tensor.add_child("occlusion", occlusion, align_dim=["B", "T"], mergeable=True)
+        return tensor
+
+    def __init__(self, x, *args, **kwargs):
+        super().__init__(x)
+
+    @classmethod
+    def from_optical_flow(
+        cls,
+        optical_flow: Flow,
+        depth: Depth,
+        next_depth: Depth,
+        intrinsic: CameraIntrinsic,
+        mode: str = "bilinear",
+    ):
+        """Create scene flow from optical flow, depth a T, depth at T + 1 and the intrinsic
+
+        Parameters
+        ----------
+        optical flow: aloscene.Flow
+            The optical flow at T.
+        depth: aloscene.Depth
+            The depth at T.
+        next_depth: aloscene.Depth
+            The depth at T + 1
+        intrinsic : aloscene.CameraIntrinsic
+            The intrinsic of the image at T.
+        """
+        has_batch = True if len(optical_flow.names) == 4 else False
+
+        if optical_flow.names != depth.names or optical_flow.names != next_depth.names:
+            raise ValueError("The optical flow, depth and next_depth must have the same names")
+
+        if optical_flow.names != ("C", "H", "W") and optical_flow.names != ("B", "C", "H", "W"):
+            raise ValueError("The optical flow must have the names (C, H, W) or (B, C, H, W)")
+
+        # Artifical batch dimension
+        optical_flow = optical_flow.batch()
+        depth = depth.batch()
+        next_depth = next_depth.batch()
+
+        H, W = depth.HW
+
+        # Compute the point cloud at T and T + 1
+        start_vector = depth.as_points3d(intrinsic).as_tensor().reshape(-1, H, W, 3).permute(0, 3, 1, 2)
+        next_vector = next_depth.as_points3d(intrinsic).as_tensor().reshape(-1, H, W, 3).permute(0, 3, 1, 2)
+
+        # Compute the position of the point cloud at T + 1
+        y_coords, x_coords = torch.meshgrid(torch.arange(H), torch.arange(W))
+        new_x = x_coords + optical_flow.as_tensor()[:, 0, :, :]
+        new_y = y_coords + optical_flow.as_tensor()[:, 1, :, :]
+
+        # Normalize the coordinates bettwen -1 and 1 and create the new points coordinates
+        new_x = new_x / W * 2 - 1
+        new_y = new_y / H * 2 - 1
+        new_coords = torch.stack([new_x, new_y], dim=3)
+
+        # Move the point cloud at T + 1 to the new position
+        end_vector = F.grid_sample(next_vector, new_coords, mode=mode, padding_mode="zeros", align_corners=True)
+
+        # Compute the scene flow
+        scene_flow_vector = end_vector - start_vector
+
+        # Create the occlusion mask
```

Comment:
Check that at least one of the augmented tensors has occlusion (optical_flow, depth, next_depth).
If it is not the case, the resulting `SceneFlow` object should have `occlusion=None`